hw_2_functions.py

```diff
@@ -0,0 +1,284 @@
+# Do you have enough toilet paper(TP) to make it through?
+# Although the number of squares per roll of TP varies significantly, we'll assume each roll has 500 sheets,
+# and the average person uses 57 sheets per day.
+
+# Create a function that will receive a dictionary with two key/values:
+# "people" ⁠— Number of people in the household.
+# "tp" ⁠— Number of rolls.
+# Return a statement telling the user if they need to buy more TP!
+
+def get_paper_forecast(h_hold, forecast_period=14):
+    """Function accepts following params:
+        1) mandatory parameters people and tp - number of people and number of TP rolls}
+        2) optional parameter forecast_period (days) that has default value of 14 days
+        """
+    sheets_per_roll = 500
+    per_person_daily_sheets = 57
+    consumption = h_hold['people'] * per_person_daily_sheets * forecast_period
+    inventory = h_hold['tp'] * sheets_per_roll
+    sheets_left = consumption - inventory
+    if sheets_left:
+        rolls = sheets_left//sheets_per_roll + 1 if (sheets_left%500) != 0 else sheets_left//500
+    print(f"For the next {forecast_period} days You need {consumption} sheets. You have {inventory} sheets.")
+    if inventory < consumption:
+        return print(f"Alarm! Run to the nearest store and buy {rolls} roll(s) of toilet paper before it's too late\n")
+    else:
+        return print(f"Keep calm and keep pooping. You have enough toilet paper for {forecast_period} day(s):)\n")
+
+
+# Testing TP rolls calculation
+print('\nTask 2. Toilet paper - to buy or not to buy\n')
+household_1 = {'people': 2, 'tp': 1}
+household_2 = {'people': 2, 'tp': 1}
+household_3 = {'people': 6, 'tp': 18}
+household_4 = {'people': 6, 'tp': 6}
+get_paper_forecast(household_1)
+get_paper_forecast(household_4, 21)
+
+
+# 3) Make a function that encrypts a given input with these steps:
+# Input: "apple"
+# Step 1: Reverse the input: "elppa"
+# Step 2: Replace all vowels using the following chart:
+# a => 0
+# e => 1
+# i => 2
+# o => 2
+# u => 3
+# # "1lpp0"
+# Example:
+# encrypt("banana") ➞ "0n0n0b"
+# encrypt("karaca") ➞ "0c0r0k"
+# encrypt("burak") ➞ "k0r3b"
+# encrypt("alpaca") ➞ "0c0pl0"
+
+
+def encrypt_text(text):
+    """Replaces every vowel letter in given text with corresponding digit from the dict"""
+    text = text[::-1]
+    vowels_dict = {'a': 0, 'e': 1, 'i': 2, 'o': 3, 'u': 4}
+    for ch in text:
+        if ch.isalpha() and ch.lower() in vowels_dict.keys():
+            text = text.replace(ch, str(vowels_dict[ch.lower()]))
+    return text
+
+
+def decrypt_text(text):
+    """Replaces every digit in the text with corresponding vowel from the dict"""
+    text = text[::-1]
+    digits_dict = {0: 'a', 1: 'e', 2: 'i', 3: 'o', 4: 'u'}
+    for ch in text:
+        if ch.isdigit() and int(ch) in digits_dict.keys():
+            text = text.replace(ch, str(digits_dict[int(ch)]))
+    return text
```

Comment:
Good idea to decrypt the the given text. However, we should admit that the initial text should contain just letters. As if it contains numbers, the decryption output may be messed.